test/e2e/scale.go

```diff
@@ -114,6 +111,9 @@ func ScaleToWithin(t *testing.T, scale int, duration time.Duration, latencies La
 			})
 
 			eg.Go(func() error {
+				ctx, cancel := context.WithTimeout(context.Background(), duration)
+				t.Cleanup(cancel)
```

Comment:
I thin you actually want defer here, since we want to kill cancel when goroutin dies

Reply:
I'm fine giving the entire test that duration.  Things like probes extend beyond the lifetime of the goroutine, so it's not alone